src/Fixer/PhpBasic/Feature/TypeHintingFixer.php

```diff
@@ -41,48 +41,35 @@ public function getDefinition()
         return new FixerDefinition(
             'We always type hint narrowest possible interface which we use inside the function or class.',
             [
-                new CodeSample(
-                    '<?php
-                        class NormalizerInterface
-                        {
-                            public function methodA();
-                        }
-                        
-                        class DenormalizerInterface
-                        {
-                            public function methodB();
-                        }
-                        
-                        class ResultNormalizer implements NormalizerInterface, DenormalizerInterface
-                        {
-                            public function methodA()
-                            {
-                                return true;
-                            }
-                            
-                            public function methodB()
-                            {
-                                return true;
-                            }
-                        }
-                    
-                        class Sample
-                        {
-                            $private $arg1;
-                        
-                            public function __construct(ResultNormalizer $arg1)
-                            {
-                                $this->arg1 = $arg1;
-                            }
-                            
-                            private function someFunction()
-                            {
-                                $this->arg1->methodA();
-                            }
-                        }
-                    '
+                new CodeSample(<<<'PHP'
```

Comment:
this example does not clear for problem that it tries to explain

Reply:
So put class definitions to comments or put previous examples to fixtures as it is now. IMO it's better to have more abstract example, eg.
```
<?php

interface InterfaceA
{
    public function a();
}

interface InterfaceB
{
    public function b();
    public function c();
}

class SomeClass implements InterfaceA, InterfaceB
{
    public function a() {}
    public function b() {}
    public function c() {}
}

class SomeManager
{
    private SomeClass $service;
    
    public function __construct(SomeClass $service)
    {
       // should be type hinted with InterfaceA instead of SomeClass
       // as we use only methods from InterfaceA
        $this->service = $service;
    }
    
    public function doA()
    {
        $this->service->a();
    }
}
````